### ISSUE TITLE
pymssql 2.2.11 does not work with TDS 7.4

### ISSUE THREAD

**Issue by zengshengliu** [pymssql/pymssql#884]
**Describe the bug**
When using pymssql 2.2.11, the library still give us error for unsupported tds version 7.4

**To Reproduce**
(I tested the following in windows)
Create a venv with python 3.10 and activate the virtual environment
```
C:\Python310\python.exe -m venv venv
.\venv\Script\activate
```
install pymssql 2.2.11
```
pip install pymssql==2.2.11
```
enter python
```
python
```
verify pymssql version
```python
import pymssql
pymssql.version_info()
```
```
"System:\nplatform.system()       : Windows\nplatform.architecture() : ('64bit', 'WindowsPE')\nPython:\nsys.version             : 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]\npymssql:\nVERSION            : (2, 2, 11)\nversion            : 2.2.11\nfull_version       : 2.2.11\ndbversion          : freetds v1.4.9\n_mssql:\nVERSION            : (2, 2, 11)\nversion            : 2.2.11\nfull_version       : 2.2.11\ndbversion          : freetds v1.4.9\nlogin_timeout      : 60\nmin_error_severity : 6\n"
```
try connecting to localhost (it doesn't exist, but if TDS verification is working, it should give us timeout instead of tds version unsupported)
```python
pymssql.connect(server='localhost', port=1433, tds_version='7.4')
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src\\pymssql\\_pymssql.pyx", line 647, in pymssql._pymssql.connect
  File "src\\pymssql\\_mssql.pyx", line 2135, in pymssql._mssql.connect
  File "src\\pymssql\\_mssql.pyx", line 655, in pymssql._mssql.MSSQLConnection.__init__
  File "src\\pymssql\\_mssql.pyx", line 1937, in pymssql._mssql._tds_ver_str_to_constant
pymssql._mssql.MSSQLException: unrecognized tds version: 7.4
```

Verify with TDS 7.3 to show the normal behavior:
```python
pymssql.connect(server='localhost', port=1433, tds_version='7.3')
```
```
Traceback (most recent call last):
  File "src\\pymssql\\_pymssql.pyx", line 647, in pymssql._pymssql.connect
  File "src\\pymssql\\_mssql.pyx", line 2135, in pymssql._mssql.connect
  File "src\\pymssql\\_mssql.pyx", line 719, in pymssql._mssql.MSSQLConnection.__init__
  File "src\\pymssql\\_mssql.pyx", line 1844, in pymssql._mssql.maybe_raise_MSSQLDatabaseException
  File "src\\pymssql\\_mssql.pyx", line 1861, in pymssql._mssql.raise_MSSQLDatabaseException
pymssql._mssql.MSSQLDatabaseException: (20009, b'DB-Lib error message 20009, severity 9:\nUnable to connect: Adaptive Server is unavailable or does not exist (localhost)\nNet-Lib error during Unknown error (10061)\n')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src\\pymssql\\_pymssql.pyx", line 653, in pymssql._pymssql.connect
pymssql._pymssql.OperationalError: (20009, b'DB-Lib error message 20009, severity 9:\nUnable to connect: Adaptive Server is unavailable or does not exist (localhost)\nNet-Lib error during Unknown error (10061)\n')
```

**Expected behavior**
TDS 7.4 should be supported according to 2.1.5 release.  (it is also supported since FreeTDS 1.0.0 version.

**Current Behavior**
`pymssql._mssql.MSSQLException: unrecognized tds version: 7.4`

**Comment by zengshengliu** [comment]
I just downloaded the source code from pypi site for version 2.2.11
https://pypi.org/project/pymssql/#files
https://files.pythonhosted.org/packages/06/b1/19a4463840c541413e58d13c1116b33669412c61528d53e02daf5e5e082b/pymssql-2.2.11.tar.gz
Looking at the _mssql.pyx file, it looks like it contains old source code.
Specifically, the code for `_tds_ver_str_to_constant(verstr) except -1:` method
```
cdef int _tds_ver_str_to_constant(verstr) except -1:
    """
        http://www.freetds.org/userguide/choosingtdsprotocol.htm
    """
    if verstr == '4.2':
        return DBVERSION_42
    if verstr == '7.0':
        return DBVERSION_70
    if verstr in ('7.1', '8.0'):
        return DBVERSION_71
    if verstr == '7.2':
        return DBVERSION_72
    if verstr == '7.3':
        return DBVERSION_73
    if verstr == '8.0':
        return DBVERSION_71
    raise MSSQLException('unrecognized tds version: %s' % verstr)
```

**Comment by termim** [comment]
It is fixed in c7584c8c164a5d92ef250e13a93f89b36faf8f5b and will be in the next release.
